### ISSUE TITLE
Missing moodle 4.4 course layout option

### ISSUE THREAD

**Issue by gustavo-magalhaes** [gjbarnard/moodle-format_grid#219]
Hi,

I have a Moodle 3.8 and I'm using the grid format in a course for HTML games. I'm trying to replicate the same settings in Moodle 4.4 (grid version 404.1.1), but I couldn't find the course layout option to display all topics on one page. This option allows you to "skip" the game section and opens it directly after clicking on each section. Am I missing a configuration to enable this option in the newer version?

I'll attach a screen shot for each version setting available. Ty for your attention!

Settings for version 3.8: ![moodle 3 8](https://github.com/user-attachments/assets/aca081bc-d2fd-4727-be97-7bdac05a53d5)

Settings for version 4.4: ![moodle 4 4](https://github.com/user-attachments/assets/417e4b93-51d0-4663-95b1-084822699919)

**Comment by gjb2048** [comment]
Use the 'Use a popup' setting:

![Screenshot 2024-09-04 135857](https://github.com/user-attachments/assets/8c0d14bf-6e68-4c73-868e-f0f455441f59)

**Comment by gustavo-magalhaes** [comment]
I've already done this, when I enable it, the only difference is that the section opens in a popup, with the other option, my file opens directly, ignoring the section and loading the resource.

Popup view:
![image](https://github.com/user-attachments/assets/80e1025d-75dd-448e-84b9-5043324d78bc)

This is how it looks on 3.8 version (by clicking in any resource, it will open the game directly, ignoring the section):
![image](https://github.com/user-attachments/assets/047d0abb-a356-4268-bb6f-c1d07b14a3e5)

**Comment by gjb2048** [comment]
But 'Show all topics on one page', never did do that.  It took you to the section where you then had to click on the link to the given module.

**Comment by gjb2048** [comment]
If you disable it then each section is on its own page.  Those are the choices as the layout setting was simply confusing.

**Comment by gustavo-magalhaes** [comment]
What I meant is that I wanted to avoid an extra click. If I use a popup or the default method, when I click on the content, I am directed to a section, and then an HTML file appears to open the game. In the old version, with the setting to show all topics on one page, clicking on the content would directly take me to the file skipping the section view.


https://github.com/user-attachments/assets/713827a8-a8ce-42bc-9722-46206074ae5f

**Comment by gjb2048** [comment]
Can you make the HTML file (mod_resource) show in the same way in the Custom Sections (was Topics) format as I suspect that the options have changed for the module, which is outside of the scope of the Grid format to control (almost but tricky).

**Comment by gjb2048** [comment]
That's not what I meant.  If the mod_folder displays 'inline' in the section in Custom Sections as you want then core Moodle still allows this and its something I need to investigate.  If it does not then (as I forget how it used to be) its a case that this was and is no longer core functionality and thus the Grid format was never actually responsible for implementing it.

**Comment by gustavo-magalhaes** [comment]
Sorry for the misunderstanding. I tested, both 3.8 moodle and 4.4 moodle, and they look like the same on the Custom Sections.

Custom Sections: ![image](https://github.com/user-attachments/assets/6e3a06d1-b319-44b5-8c9a-534222ecb6a5)

Is there anything I sould try here so I can highlight to you?

Btw sorry for extending this question that much.

**Comment by gjb2048** [comment]
What's the screen shot for M3.8 please?

Have you tried putting the HTML in the section summary?

**Comment by gustavo-magalhaes** [comment]
This is the view with the grid: ![image](https://github.com/user-attachments/assets/e15527d9-8af7-4f83-abe0-3f8f136b3ad3)

Custom Section (Topics): ![image](https://github.com/user-attachments/assets/c1de763f-bb58-4251-9adc-a5f95d8e08d0)

Interestingly, when I switch the course layout option to show all topics on one page, version 3.8 behaves the same way as 4.4, but version 4.4 doesn’t have this setting (first message have the setting view for each version). 

Regarding the summary, it remains unchanged.

**Comment by gjb2048** [comment]
But you said that M3.8 was working with 'show all topics on one page'?  Or are you saying that Grid in M3.8 only works with 'show all topics on one page' and Topics only works in M3.8 when set to the other multipage option.

And, no, with "Have you tried putting the HTML in the section summary?" I mean put the markup in the section summary in the editor when editing using HTML mode.

**Comment by gustavo-magalhaes** [comment]
I will try to rephrase my question, I may have expressed myself incorrectly:

Today, in Moodle 3.8, which is my production version, I have a course that I use as a "gaming platform." I use the File activity type to upload the game folder and set the HTML file as the main file. I use the Grid Format compatible with the Moodle version to ensure everything works correctly. One of the configurations of this format is a course layout option, which is set to "show all topics on one page."

In Moodle 4.4, I installed the Grid Format to make my "gaming platform" work and noticed that the course layout option was missing. When I clicked on one of the icons (resources), I was redirected to a page that opened the section first (something that did not happen in Moodle 3.8). After testing a few things, I noticed that in Moodle 3.8, if I change the course layout option to show one topic per page, I lose the functionality to open the HTML file directly and am also redirected to the section.

In short, I wanted to know if in Moodle 4.4 it is possible to configure the course layout to show all content on one page, just like it is possible in 3.8.

Please excuse my ignorance on some points, I just started developing in Moodle, and I am still a bit confused xD.

**Comment by gjb2048** [comment]
But that's not the question.  The question is if it is possible to show the contents of the mod_folder instance without first clicking on the url first.

I no longer run a Moodle 3.8 instance and it will take me time to get one up and running again when I can.  Pragmatically its so far in the past that I've forgotten how it works.

However, I believe that the workaround is to put the content (HTML) in the section summary.

**Comment by gjb2048** [comment]
But it is also a possibility that the functionality you thought was the Grid format was actually indeed core functionality and thus won't actually work on M4.4 anyway.

**Comment by gjb2048** [comment]
Revisiting your video at 0:05 that is 'Show one topic per page' not 'Show all topics on one page' in the course settings because of the appearance of the section navigation at the bottom.

**Comment by gustavo-magalhaes** [comment]
> But it is also a possibility that the functionality you thought was the Grid format was actually indeed core functionality and thus won't actually work on M4.4 anyway.

If I set format course to Custom Sections, the course layout option to show all sections on one page is there: ![image](https://github.com/user-attachments/assets/4b50bacd-31c7-4ba5-806a-cb198fb4cc98)

**Comment by gjb2048** [comment]
No, I mean not the course layout option, but that when 'Show one section per page' is set in the Custom Sections format, that the File resource being a HTML file will not show inline but instead have a link.

**Comment by gustavo-magalhaes** [comment]
> Revisiting your video at 0:05 that is 'Show one topic per page' not 'Show all topics on one page' in the course settings because of the appearance of the section navigation at the bottom.

Actually, it's set to "show all sections on one page"

https://github.com/user-attachments/assets/9c152e72-5aba-44a3-b65b-1dc490cd2b39

**Comment by gjb2048** [comment]
Ok, I'm going to have to find a moment to install M3.8 etc.

In the mean time: "I believe that the workaround is to put the content (HTML) in the section summary."

**Comment by gustavo-magalhaes** [comment]
Is it possible to upload my game files to the summary? This is a structure for a game:

![image](https://github.com/user-attachments/assets/5ade2025-71bf-4558-9b48-b37a8f87cf55)

**Comment by gjb2048** [comment]
I don't think so.

**Comment by gjb2048** [comment]
@gustavo-magalhaes Ok, do you have custom code that disables the 'modal' as in M3.8, I can't replicate what you're showing me:


https://github.com/user-attachments/assets/1dc732fe-cfce-406f-8948-4a248ac855e9

**Comment by gustavo-magalhaes** [comment]
That's a possibility, there were other people working on this moodle before me. In fact, I searched all over the structure trying to find changes, but didn't, that's why I oppened this topic. Thank you for your time and effort on this case.

**Comment by gjb2048** [comment]
@gustavo-magalhaes As far as I can determine, you have some custom code that is jumping from the section to the module.

**Comment by gustavo-magalhaes** [comment]
@gjb2048, sorry for the late reply. I tested again on a local Moodle 3.8, and you're right there is, in fact, a customization on the plugin. Unfortunately, I couldn't find where, but I'll keep trying. Thanks again for your help.

**Comment by gustavo-magalhaes** [comment]
@gjb2048 FYI, it was really necessary to customize some behaviors on Moodle, it's working as expected now.

**Comment by gjb2048** [comment]
@gustavo-magalhaes Thanks for letting me know.  Can this issue be closed?
